src/main/java/com/internet/shop/dao/jdbc/UserDaoJdbcImpl.java

```diff
@@ -0,0 +1,201 @@
+        User user = null;
+        try (Connection connection = ConnectionUtil.getConnection()) {
+            PreparedStatement statement = connection.prepareStatement(query);
+            statement.setLong(1, id);
+            ResultSet resultSet = statement.executeQuery();
+            if (resultSet.next()) {
+                user = getUserFromResultSet(resultSet);
+            }
+        } catch (SQLException ex) {
+            throw new DataProcessingException("Can't get user with id = " + id, ex);
+        }
+
+        if (user != null) {
+            user.setRoles(getUserRolesFromResultSet(user.getId()));
+        }
+        return Optional.ofNullable(user);
+    }
+
+    @Override
+    public List<User> getAll() {
+        String query = "SELECT * FROM users WHERE deleted = FALSE;";
+        List<User> users = new ArrayList<>();
+        try (Connection connection = ConnectionUtil.getConnection()) {
+            PreparedStatement statement = connection.prepareStatement(query);
+            ResultSet resultSet = statement.executeQuery();
+            while (resultSet.next()) {
+                User user = getUserFromResultSet(resultSet);
+                users.add(user);
+            }
+        } catch (SQLException ex) {
+            throw new DataProcessingException("Can't get all users", ex);
+        }
+
+        for (User user: users) {
+            user.setRoles(getUserRolesFromResultSet(user.getId()));
+        }
+        return users;
+    }
+
+    @Override
+    public User update(User user) {
+        String query = "UPDATE users SET user_name = ?, login = ?, password = ? "
+                + "WHERE user_id = ? AND deleted = FALSE;";
+        try (Connection connection = ConnectionUtil.getConnection()) {
+            PreparedStatement statement = connection.prepareStatement(query);
+            statement.setString(1, user.getName());
+            statement.setString(2, user.getLogin());
+            statement.setString(3, user.getPassword());
+            statement.setLong(4, user.getId());
+            statement.executeUpdate();
+        } catch (SQLException ex) {
+            throw new DataProcessingException("Can't update user with id = "
+                    + user.getId(), ex);
+        }
+
+        removeUserRoles(user.getId());
+        insertUserRoles(user);
+        return user;
+    }
+
+    @Override
+    public boolean delete(Long id) {
+        String query = "UPDATE users SET deleted = TRUE WHERE user_id = ?;";
+        int itemsDeleted = 0;
+        try (Connection connection = ConnectionUtil.getConnection()) {
+            PreparedStatement statement = connection.prepareStatement(query);
+            statement.setLong(1, id);
+            itemsDeleted = statement.executeUpdate();
+        } catch (SQLException ex) {
+            throw new DataProcessingException("Can't delete product with id = " + id, ex);
+        }
+        return (itemsDeleted == 1 && removeUserRoles(id));
```

Comment:
@RomanDubovskyi And what about removing user roles from users_roles table?

Reply:
while deleting it's enough to mark user as deleted, don't remove his relation with roles